crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -80,15 +79,22 @@ pub fn service_monitor_client_config(tedge_config: &TEdgeConfig) -> Result<Confi
     let device_name = tedge_config.device.id.try_read(tedge_config)?.to_string();
     let service_type = tedge_config.service.ty.clone();
 
+    // from this level we don't have access to the entity store and registered main device so best
+    // we can do for now is just guess it
+    // TODO: fix this, preferably use a HealthMonitorActor
+    let service_external_id = format!("{device_name}:device:main:service:tedge-mapper-c8y");
+
+    let last_will_message = c8y_api::smartrest::inventory::service_creation_message(
+        service_external_id.as_str(),
+        "tedge-mapper-c8y",
+        service_type.as_str(),
+        "down",
+        &[],
+    );
+
     let mqtt_config = tedge_config
         .mqtt_config()?
         .with_session_name("last_will_c8y_mapper")
-        .with_last_will_message(service_monitor_status_message(
```

Comment:
So the call to `service_monitor_status_message` has been replaced by a more generic one .i.e. `service_creation_message`. This seems okay at first (both are 102 smartrest messages), but are you aware that this last will message has a very specific purpose (telling c8y that thin-edge is down) that is not the same as the other health checks (checking that all the components are up).

Reply:
> this last will message has a very specific purpose (telling c8y that thin-edge is down) that is not the same as the other health checks (checking that all the components are up).

I was not aware of the distinction, in fact I was thinking of how to remove this custom last will message for the mapper and handle it more like other services.

Is this change incorrect or does it significantly change the behaviour in some way? If so, then I can fix it and perhaps make a follow-up test if some corner case that we don't currently test for has been violated by this change.

Reply:
So, we need to double check.

@PradeepKiruvale can you check that this PR doesn't break the status reported by thin-edge on c8y?

Reply:
> I didn't get the point of contention here. We're still passing the static `down` message here for the mapper service, with the mapper name, as it used to be, just with a different API. The new API doesn't differentiate between `up` or `down` or any status value for that matter. So, the replacement feels the same. Am I missing something subtle here?

My question is not about the up or down status, but about the smartrest message that is sent. Is the same message type used both to declare a new service and to notify a service status?

Reply:
Are these changes the expected ones?
* New way to name a service:  `demo-device-777_tedge-mapper-c8y` -> `demo-device-777:device:main:service:tedge-mapper-c8y`
* New `104` messages on sub topics.

On start (new version):

```
[c8y/s/us] 102,demo-device-777:device:main:service:tedge-mapper-c8y,service,tedge-mapper-c8y,up
[c8y/s/us/demo-device-777:device:main:service:tedge-mapper-c8y] 104,up
```

When c8y-mapper exit (new version):
```
[c8y/s/us] 102,demo-device-777:device:main:service:tedge-mapper-c8y,service,tedge-mapper-c8y,down
```

On start (previous version):
```
[c8y/s/us] 102,demo-device-777_tedge-mapper-c8y,"service",tedge-mapper-c8y,"up"
```

On stop (previous version):
```
[c8y/s/us] 102,demo-device-777_tedge-mapper-c8y,"service",tedge-mapper-c8y,"down"
```

Reply:
Changing service names is a bit annoying on c8y as of now I have two c8y-mapper on my console ;-)

But this is an expected drawback of the new naming scheme.

Reply:
I verified that the `last will` message is getting translated to the `service` message and pushed to the c8y cloud by `tedge-mapper-c8y`.

```
[te/device/main/service/tedge-mapper-c8y/status/health] {"pid":404449,"status":"down"}
[c8y/s/us] 102,wsl-ubuntu:device:main:service:tedge-mapper-c8y,service,tedge-mapper-c8y,down

```

Reply:
> Are these changes the expected ones?
> 
>     * New way to name a service:  `demo-device-777_tedge-mapper-c8y` -> `demo-device-777:device:main:service:tedge-mapper-c8y`

I followed the new C8y device and service topics described in [_Cumulocity Mapper_ documentation page](https://thin-edge.github.io/thin-edge.io/next/references/mappers/c8y-mapper/#child-device-service)



> New `104` messages on sub topics.

Yes, I changed to `104` because it doesn't need a `serviceType`, which simplifies things a bit.

Reply:
> However, the behavior of `104` message is buggy.

This is due to the `tedge` to `te` converter. As I'm testing this PR on a machine where old health messages are retained on `tedge/health/+`, these messages are translated by `tedge_to_te_converter` on agent starts messing up the messages sent on `te/+/+/+/+/status/+` by the thin-edge daemons. Only the later ones are correct.

Since, this PR updates *all* thin-edge daemons to use `te/+/+/+/+/status/+` , the `tedge_to_te_converter`  must stop converting the former messages:


```
diff --git a/crates/core/tedge_agent/src/tedge_to_te_converter/converter.rs b/crates/core/tedge_agent/src/tedge_to_te_converter/converter.rs
index 1ebb63e21f..d0753079ea 100644
--- a/crates/core/tedge_agent/src/tedge_to_te_converter/converter.rs
+++ b/crates/core/tedge_agent/src/tedge_to_te_converter/converter.rs
@@ -30,9 +30,6 @@ impl TedgetoTeConverter {
             }
             topic if topic.name.starts_with("tedge/events") => self.convert_event(message),
             topic if topic.name.starts_with("tedge/alarms") => self.convert_alarm(message),
-            topic if topic.name.starts_with("tedge/health") => {
-                self.convert_health_status_message(message)
-            }
             _ => vec![],
         }
     }
(
```

Reply:
> > However, the behavior of `104` message is buggy.
> 
> This is due to the `tedge` to `te` converter.

Indeed this behaviour is buggy and should be corrected. I will remove `tedge_to_te_converter` mapping of health status as part of this PR.

Reply:
Thanks to @PradeepKiruvale and @Bravo555 for checking behavior and expectation on `102` and `104`.

Once `tedge_to_te_converter` fixed, the status are propagated as expected.